tested/dsl/translate_parser.py

```diff
@@ -587,24 +631,34 @@ def _convert_testcase(testcase: YamlDict, context: DslContext) -> Testcase:
     else:
         the_description = None
 
-    return Testcase(
-        description=the_description,
-        input=the_input,
-        output=output,
-        link_files=context.files,
-        line_comment=line_comment,
+    return DataWithMessage(
+        data=Testcase(
+            description=the_description,
+            input=the_input,
+            output=output,
+            link_files=context.files,
+            line_comment=line_comment,
+        ),
+        messages=messages,
     )
 
 
-def _convert_context(context: YamlDict, dsl_context: DslContext) -> Context:
+def _convert_context(
+    context: YamlDict, dsl_context: DslContext
+) -> DataWithMessage[Context, ExtendedMessage]:
     dsl_context = dsl_context.deepen_context(context)
     raw_testcases = context.get("script", context.get("testcases"))
     assert isinstance(raw_testcases, list)
-    testcases = _convert_dsl_list(raw_testcases, dsl_context, _convert_testcase)
-    return Context(testcases=testcases)
+    data_w_messages = _convert_dsl_list(raw_testcases, dsl_context, _convert_testcase)
```

Comment:
Don't shorten the `with`

I would even prefer if you just kept the name `testcases`

This feedback is applicable multiple times in this file